Framework/Algorithms/src/AbsorptionCorrection.cpp

```diff
@@ -420,6 +420,7 @@ void AbsorptionCorrection::calculateDistances(const IDetector &detector, std::ve
 /// instruments
 double AbsorptionCorrection::doIntegration(const double linearCoefAbs, const std::vector<double> &L2s,
                                            const size_t startIndex, const size_t endIndex) const {
+  // cppcheck-suppress knownConditionTrueFalse
```

Comment:
This is from set 7.

Reply:
Sorry for missing off an explanation.

If I recall correctly `endIndex` and `startIndex` can both be input (as you can see in the function), so this condition could be either `True` or `False`. However, currently the function is only used with hardcoded values, so we believe this gets optimized out into `if (const - const > const)`, and therefore `if const`.

If a developer was to make changes that use this function differently in the future, this check would still be needed, hence the suppression.